zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/archiver/OpenSearchRepository.java

```diff
@@ -0,0 +1,125 @@
+import org.opensearch.client.opensearch._types.FieldValue;
+import org.opensearch.client.opensearch._types.Time;
+import org.opensearch.client.opensearch._types.aggregations.CalendarInterval;
+import org.opensearch.client.opensearch._types.query_dsl.QueryBuilders;
+import org.opensearch.client.opensearch._types.query_dsl.TermsQuery;
+import org.opensearch.client.opensearch.core.DeleteByQueryRequest;
+import org.opensearch.client.opensearch.core.DeleteByQueryResponse;
+import org.slf4j.Logger;
+
+public final class OpenSearchRepository extends NoopArchiverRepository {
+  private static final String DATES_AGG = "datesAgg";
+  private static final String INSTANCES_AGG = "instancesAgg";
+  private static final String DATES_SORTED_AGG = "datesSortedAgg";
+  private static final Time REINDEX_SCROLL_TIMEOUT = Time.of(t -> t.time("30s"));
+  private static final long AUTO_SLICES = 0; // see OS docs; 0 means auto
+
+  private final int partitionId;
+  private final ArchiverConfiguration config;
+  private final RetentionConfiguration retention;
+  private final String processInstanceIndex;
+  private final String batchOperationIndex;
+  private final OpenSearchAsyncClient client;
+  private final Executor executor;
+  private final CamundaExporterMetrics metrics;
+  private final Logger logger;
+
+  private final CalendarInterval rolloverInterval;
+
+  public OpenSearchRepository(
+      final int partitionId,
+      final ArchiverConfiguration config,
+      final RetentionConfiguration retention,
+      final String processInstanceIndex,
+      final String batchOperationIndex,
+      @WillCloseWhenClosed final OpenSearchAsyncClient client,
+      final Executor executor,
+      final CamundaExporterMetrics metrics,
+      final Logger logger) {
+    this.partitionId = partitionId;
+    this.config = config;
+    this.retention = retention;
+    this.processInstanceIndex = processInstanceIndex;
+    this.batchOperationIndex = batchOperationIndex;
+    this.client = client;
+    this.executor = executor;
+    this.metrics = metrics;
+    this.logger = logger;
+
+    rolloverInterval = mapCalendarInterval(config.getRolloverInterval());
+  }
+
+  @Override
+  public CompletableFuture<Void> deleteDocuments(
+      final String sourceIndexName,
+      final String idFieldName,
+      final List<String> processInstanceKeys) {
+    final TermsQuery termsQuery = buildIdTermsQuery(idFieldName, processInstanceKeys);
+    final var request =
+        new DeleteByQueryRequest.Builder()
+            .index(sourceIndexName)
+            .slices(AUTO_SLICES)
+            .conflicts(Conflicts.Proceed)
+            .query(q -> q.terms(termsQuery))
+            .build();
+
+    final var timer = Timer.start();
+    return sendRequestAsync(() -> client.deleteByQuery(request))
+        .whenCompleteAsync((ignored, error) -> metrics.measureArchiverDelete(timer), executor)
+        .thenApplyAsync(DeleteByQueryResponse::total, executor)
```

Comment:
This is the total of records processed correct? Are we using this in the future for metrics? In the documentation I also saw another response field which is delete -> "Total number of documents deleted.". I am not sure what is exactly the difference between the two in this case. 🤔